source/particle/generator/quadrature_points.cc

```diff
@@ -43,8 +43,13 @@ namespace aspect
                                 quadrature_formula,
                                 update_quadrature_points);
 
-
+#if DEAL_II_VERSION_GTE(9,1,0)
+        MPI_Scan(&n_particles_to_generate, &prefix_sum, 1, DEAL_II_PARTICLE_INDEX_MPI_TYPE, MPI_SUM, this->get_mpi_communicator());
+#elif DEAL_II_VERSION_GTE(9,0,0)
+        MPI_Scan(&n_particles_to_generate, &prefix_sum, 1, PARTICLE_INDEX_MPI_TYPE, MPI_SUM, this->get_mpi_communicator());
+#else
         MPI_Scan(&n_particles_to_generate, &prefix_sum, 1, ASPECT_PARTICLE_INDEX_MPI_TYPE, MPI_SUM, this->get_mpi_communicator());
```

Comment:
how about you define ASPECT_PARTICLE_INDEX_MPI_TYPE to be the correct thing so that you can remove all these ifdefs here?

Reply:
Hmm, currently I only use this type in two locations, and if I define ASPECT_PARTICLE_INDEX_MPI_TYPE I need an ifdef there anyway (two distinguish between PARTICLE_INDEX_MPI_TYPE for deal 9.0 and DEAL_II_PARTICLE_INDEX_MPI_TYPE for 9.1). Plus then I need to keep the definition of ASPECT_PARTICLE_INDEX_MPI_TYPE, while this ifdef can go away once we require 9.1. So I would prefer to keep it in the current form, is that ok with you?

Reply:
alright, makes sense.